### PR TITLE
Fixed types (monad to monoid), switch the order of monoids and an exa…

### PR DESCRIPTION
**Pull request by ghost**
…mple, added solutions of categories, isomorphisms and monomorphisms

### CODE REVIEW

File: tex/CT4P.tex [hunk 1]
```diff
@@ -495,7 +495,7 @@ \section{Categories}
 The proof that left unit law and associator law hold are analogous.
 \end{exa}
 
-\begin{exer}
+\begin{exer}\label{exer:post_antisymmetry}
```
Comment (benediktahrens) [inline tex/CT4P.tex:498]: Here and elsewhere, I suggest we add a pointer to the solution, like this (or in a better format):
```suggestion
\begin{exer}[See \cref{label-of-solution}]\label{exer:post_antisymmetry}
```
It is best to use the same label as the exercise, but with the prefix "sol:" or "solution:".

File: tex/CT4P.tex [hunk 2]
```diff
@@ -844,19 +876,19 @@ \subsection{Monomorphisms and Epimorphisms}
...
 
-\begin{exer}
-  In the category of sets, show that any section is injective.
+\begin{exer}\label{exer:sections_in_set_injective}
+  In the category of sets, show that if $(s,r)$ is a section-retraction pair, then is the section $s$ injective.
   Hint: you can use \cref{ex:mono-inj}.
 \end{exer}
 \begin{exer}
-  In the category of sets, show that any retraction is surjective.
+  In the category of sets, show that if $(s,r)$ is a section-retraction pair, then is the retraction $r$ surjective.
```
Comment (benediktahrens) [inline tex/CT4P.tex]:
```suggestion
  In the category of sets, show that if $(s,r)$ is a section-retraction pair, then the retraction $r$ is surjective.
```

File: tex/CT4P.tex [hunk 3]
```diff
@@ -844,19 +876,19 @@ \subsection{Monomorphisms and Epimorphisms}
...
     Hint: you can use \cref{ex:epi-surj}.
 \end{exer}
 
 \begin{exer}\label{exer:iso_to_monoepi} Show that any isomorphism $f : a\cong b$ (in some arbitrary category $\CC$) is both a monomorphism and an epimorphism.
 \end{exer}
 
-\begin{exer} Show that the converse of \cref{exer:iso_to_monoepi} does not hold in general, i.e. give an example of a category where there exists a morphism which is both an epi- and a monomorphism, but which is not an isomorphism.\\
+\begin{exer}\label{exer:counterexample_monoepi_not_iso} Show that the converse of \cref{exer:iso_to_monoepi} does not hold in general, i.e. give an example of a category where there exists a morphism which is both an epi- and a monomorphism, but which is not an isomorphism.\\
```
Comment (benediktahrens) [inline tex/CT4P.tex]:
```suggestion
\begin{exer}\label{exer:counterexample_monoepi_not_iso} Show that the converse of \cref{exer:iso_to_monoepi} does not hold in general, i.e. give an example of a category where there exists a morphism which is both an epi- and a monomorphism, but which is not an isomorphism.

```

File: tex/solutions.tex [hunk 4]
```diff
@@ -0,0 +1,280 @@
+\section{Solutions}
+\label{sec:solutions}
+
+\begin{solution}[to \cref{exer:post_antisymmetry}] 
+The inequality $x\leq y$ (resp. $y\leq x$) means that we have a (unique) morphism from $f\in \Hom{x}{y}$ (resp. $g\in \Hom{y}{x}$). Consequently, we get a \textit{loop} $\co{f}{g} \in \Hom{x}{x}$. Since the hom-sets are either empty or a singleton, we have $\co{f}{g} = \Id[x]$. Hence, antisymmetry means that if $\Id[X] = \co{f}{g}$ for some $f\in \Hom{x}{y}$ and $g\in \Hom{y}{x}$, we must have 
+$$x=y,\quad f = \Id[x] = g.$$
```
Comment (benediktahrens) [inline tex/solutions.tex]: Double-dollars are not recommended, they don't produce good spacing.
```suggestion
\[x=y,\quad f = \Id[x] = g.\]
```

File: tex/solutions.tex [hunk 5]
```diff
@@ -0,0 +1,280 @@
...
+\begin{eqnarray*}
+g_1 =& \co{g_1}{\Id[Y]} & \text{ by unit law},\\ 
+	=& \co{g_1}{\co{f}{h}} & \text{ since $f$ section},\\ 
+	=& \co{\co{g_1}{f}}{h} & \text{ by associativity},\\
+	=&  \co{\co{g_2}{f}}{h} & \text{ by assumption},\\
+	=& \co{g_2}{\co{f}{h}} & \text{ by associativity},\\ 
+	=& \co{g_2}{\Id[Y]} & \text{ since $f$ section},\\
+	=& g_2 & \text{ by unit law}.
+\end{eqnarray*}
+Notice that this proof shows that in an arbitrary category, a section is always a monomorphism.
```
Comment (benediktahrens) [inline tex/solutions.tex:247]: This remark is maybe better positioned after the exercise?

**Review by benediktahrens — APPROVED** [review]
LGTM modulo small comments

File: tex/solutions.tex (hunk 5 above)
Comment (ghost) [reply tex/solutions.tex:247]: But this remark only makes sense when considering the solution, so after the exercise in the solution file or in the CT4P file after the exercise?

Comment (benediktahrens) [reply tex/solutions.tex:247]: After the exercise, you can say "The solution given in \cref{solution:foobar} actually works for any category. Hence we can state the following lemma:"

This is just a suggestion, anything is good with me. But I think it is best if general statements are given in the main text, not in the solutions.

**Pull request merged**